src/lib/util/feature-evaluator/strategy/application-hostname-strategy.ts

```diff
@@ -0,0 +1,27 @@
+import { hostname } from 'os';
+import { Strategy } from './strategy';
+
+export default class ApplicationHostnameStrategy extends Strategy {
+    private hostname: string;
+
+    constructor() {
+        super('applicationHostname');
+        this.hostname = (
+            process.env.HOSTNAME ||
+            hostname() ||
+            'undefined'
+        ).toLowerCase();
+    }
+
+    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+    isEnabled(parameters: { hostNames: string }): boolean {
+        if (!parameters.hostNames) {
+            return false;
+        }
+
+        return parameters.hostNames
+            .toLowerCase()
+            .split(/\s*,\s*/)
+            .includes(this.hostname);
```

Comment:
for posterity: collect the parameter names and context names for default strategies

Reply:
They are:

| strategy                | parameters property names          | context field   |
|-------------------------|------------------------------------|-----------------|
| application hostname    | `hostNames`                        | `hostname`      |
| flexible rollout        | `groupId`, `rollout`, `stickiness` | `featureToggle` |
| gradual rollout random  | `percentage`                       |                 |
| gradual rollout session | `percentage`, `groupId`            | `sessionId`     |
| gradual rollout session | `percentage`, `groupId`            | `userId`        |
| remote address          | `IPs`                              | `remoteAddress` |
| user id                 | `userIds`                          | `userId`        |